src/library/layer.lua

```diff
@@ -0,0 +1,87 @@
+--[[
+$module Layer
+]]
+
+layer = {}
+
+--[[
+% copy(source_layer, destination_layer)
+
+Duplicates the notes from the source layer to the destination. The source layer remains untouched.
+
+@ source_layer number the number (1-4) of the layer to duplicate
+@ destination_layer number the number (1-4) of the layer to be copied to
+]]
+function layer.copy(source_layer, destination_layer)
+    local region = finenv.Region()
```

Comment:
Can we move `region` to be an argument to the function? For instance:

```lua
layer.copy(region, source_layer, destination_layer)
```

In general when writing functions—especially when it's part of a standard library—it's best to make sure the function's actions depend solely on the given arguments.

Why? There are many reasons but the most applicable one is predictability.

For instance, if you see this in the codebase `layer.copy(1, 2)`, you might be confused what _exactly_. Is this on the first staff? The first measure? The entire piece? By explicitly passing in a region `layer.copy(region, 1, 2)` you instantly know that you're copying layer 1 into layer 2 for the specified region.

If you want to read up more on this, learn about coupling and cohesion, as well as pure functions:

- [Coupling and cohesion](https://youtu.be/eiDyK_ofPPM)
- [Pure functions](https://youtu.be/4IIWib5MZKg)

_Note that pure functions aren't always strictly what we want, there are real reasons why you wouldn't want to use them (e.g., if you want to change state, i.e., actually copying music). But the idea of only relying on the inputs is generally a good idea since it lowers coupling a lot._